data/transactions/logic/opcodes_test.go

```diff
@@ -35,6 +35,12 @@ func TestOpSpecs(t *testing.T) {
 func TestOpcodesByVersion(t *testing.T) {
 	t.Parallel()
 
+	// To test the case where a newer version opcode is before an older version
+	// Change the order of opcode 0x01 so that version 2 comes before version 1
+	tmp := OpSpecs[1]
+	OpSpecs[1] = OpSpecs[4]
+	OpSpecs[4] = tmp
+
```

Comment:
I think this should be a separate test case with the shared implementation with the current one

Reply:
done.